db/rocksdb_ethereumtype.go

```diff
@@ -17,14 +19,103 @@ import (
 const InternalTxIndexOffset = 1
 const ContractIndexOffset = 2
 
+type AggregateFn = func(*big.Int, *big.Int)
+
+type Ids []big.Int
+
+func (s *Ids) sort() bool {
+	sorted := false
+	sort.Slice(*s, func(i, j int) bool {
+		isLess := (*s)[i].Cmp(&(*s)[j]) == -1
+		if isLess == (i > j) { // it is necessary to swap - (id[i]<id[j] and i>j) or (id[i]>id[j] and i<j)
+			sorted = true
+		}
+		return isLess
+	})
+	return sorted
+}
+
+func (s *Ids) search(id big.Int) int {
+	// attempt to find id using a binary search
+	return sort.Search(len(*s), func(i int) bool {
+		cmp := (*s)[i].Cmp(&id)
```

Comment:
How about using `CmpAbs`. I do not think there are any negative ids and `CmpAbs` is slightly more performant.